packages/vue-components/src/Overlay.vue

```diff
@@ -57,7 +57,12 @@ export default {
   },
   methods: {
     toggleNavMenu() {
-      if (!this.show) { publish('closeOverlay'); }
+      if (!this.show) {
+        publish('closeOverlay');
+        document.body.style.overflow = 'hidden';
+      } else {
+        document.body.style.removeProperty('overflow');
+      }
```

Comment:
This extra block is needed so that when the `overlay` is opened, it prevents the header from hiding as a result of scroll chaining scrolling the body.

Reply:
Can I just clarify what you mean by this? 

![Screen Shot 2021-02-11 at 12 01 31 AM](https://user-images.githubusercontent.com/37920115/107536040-4ea07000-6bfc-11eb-8bbd-c5d0c6da8d26.png)

Is it something like the screenshot above where the navbar is hidden partially when you scroll down?

Reply:
> Would it be possible to have a transition effect from no header <-> show header? Without it, I feel that the transition might be a little abrupt. What do you think @jonahtanjz ?
> 
> ![Screen-Recording-2021-02-10-at-11 59 55-PM](https://user-images.githubusercontent.com/16506539/107535996-3d576380-6bfc-11eb-8aa4-8ce59600fe72.gif)

I agree with @raysonkoh on this. Actually, I was thinking if the overlay effect that I mentioned above would be a suitable "transition" effect? I don't see many transition effect options that are suitable for this, to be honest :O

Reply:
I've dabbled around the transitions and here's something I came up with. It's not the best solution but I was hoping that it may help you come up with better ideas :-)

https://user-images.githubusercontent.com/37920115/107546913-d8096f80-6c07-11eb-878f-5897fc650336.mov

Reply:
Thanks for the suggestions @wxwxwxwx9 

> Can I just clarify what you mean by this?

This is when either the site or page nav is opened and the user scroll to the end, if the user continues scrolling, it will scroll the main body as well. Can read more about it here: https://developer.mozilla.org/en-US/docs/Web/CSS/overscroll-behavior. You can also try it out on the current user guide's site nav: https://markbind-master.netlify.app/userguide/gettingstarted.

> I agree with @raysonkoh on this. Actually, I was thinking if the overlay effect that I mentioned above would be a suitable "transition" effect? I don't see many transition effect options that are suitable for this, to be honest :O

Yup agree with you and @raysonkoh on the transition and I was planning to implement it but was kind of stuck on this as the standard css transitions doesn't really work well here. Will have to find alternatives.

> I've dabbled around the transitions and here's something I came up with using max-height css transition. It's not the best solution but I was hoping that it may help with the discussion :-)

This actually looks good. Thanks for looking into this 👍. I'm ok with using this. @raysonkoh what do you think about this?

Reply:
Yup I think it looks good. We can go with this. 👍

Reply:
> This is when either the site or page nav is opened and the user scrolls to the end, if the user continues scrolling, it will scroll the main body as well. Can read more about it here: https://developer.mozilla.org/en-US/docs/Web/CSS/overscroll-behavior. You can also try it out on the current user guide's site nav: https://markbind-master.netlify.app/userguide/gettingstarted.

Ah, I see! Understood. I tried out another solution: place `overscroll-behaviour-y: none` under `.nav-menu`. It's working on my side. Can you try if it works for you? I think it's a more succinct solution but I'm not sure if it would interfere with the features you have implemented so far (e.g. siteNav, navBar, and this PR). What do you think? :-)

Reply:
Yup agree that using this css would be much more succinct solution as I initially wanted to use this as well. However, for some reason, it does not work very well when the nav menus are not scrollable. One example is https://markbind-master.netlify.app/userguide/tweakingthepagestructure, able to scroll the body of page nav when scrolling the ride side of the overlay. This is more obvious on larger screen sizes.

https://user-images.githubusercontent.com/47470981/107728574-6b869180-6d29-11eb-89b1-a4d9990b1b20.mp4

Reply:
Ahh, I see! Yup, it's not working on my side as well. I think it's because when nav menus are not scrollable (not enough content), overscroll behaviour does not kick in since there's nothing to scroll. 

In that case, I think your solution of making the body hidden when overlay is opened makes sense, since we don't want them to be able to scroll the body content when they have overlay opened (since the focus should be on overlay). 

For easy reference in the future as to why we add this block, I think we should add some comments to explain the rationale. Maybe something like `document.body.style.overflow = 'hidden'; // to prevent scrolling of the body when overlay is overscrolled`.

Reply:
how about just `@scroll.stop` on the root element of the overlay to prevent the scroll event from bubbling up?